### PR TITLE
Feature/farming page 15

### PR DESCRIPTION
**Pull request by natonnelier**
farming-page

Changes:

- Right align the value and left align the token name in "Underlying Token" column
- Staked, in USD - move the column next to TVL
- Active campaigns displayed by default
- Select box added to switch between Active campaigns and Expired campaigns.

### CODE REVIEW

File: src/components/FarmingList/index.js [hunk 1]
```diff
@@ -223,46 +223,53 @@ function FarmingList({ campaigns, color, disbaleLinks, maxItems = 10 }) {
...
                   nativeCurrencyWrapper.symbol ===
-                  pairData.stakablePair.token1.symbol
+                    pairData.stakablePair.token1.symbol
                     ? nativeCurrency
                     : pairData.stakablePair.token1.symbol
                 }
                 maxCharacters={below600 ? 8 : 16}
                 adjustSize={true}
                 link={true}
+                style={{ flexBasis: "30%", textAlign: "left" }}
```
Comment (Mi-Lan) [inline src/components/FarmingList/index.js:272]: I don't think its good to use inline styles in this way flex-basis in particular it would be good if we can do it some other way since its not DRY and easy to mainatin...

File: src/pages/FarmingPage.js [hunk 2]
```diff
@@ -9,11 +9,71 @@ import {
...
+    "active": "Active Campaigns"
+  }
+
+  const handleFilterCampaigns = (selected) => {
+    const status = Object.keys(options).find(key => options[key] === selected);
+    filterCampaigns(status);
+  }
+
+  const memoizedActiveCampaigns = memoize(activeCampaigns);
+  const memoizedExpiredCampaigns = memoize(expiredCampaigns);
```
Comment (Mi-Lan) [inline src/pages/FarmingPage.js:76]: 1) I think this filtering should be done in useLiqudityMiningCampaignData and served directly to component since its only use case for given hook
2) Subgraph query needs to be changed to include expired campaigns since it just fetches active campaigns
Screenshot 2021-11-12 at 18 32 56

**Review by Mi-Lan — CHANGES_REQUESTED** [review]
Left some comments. Good job overall!

**Comment by Mi-Lan** [comment]
Screenshot 2021-11-12 at 18 37 58
Also this compile error

**Comment by natonnelier** [comment]
> Left some comments. Good job overall!

fixes added @Mi-Lan ! Please take a look

File: src/pages/FarmingPage.js [hunk 3]
```diff
@@ -1,19 +1,43 @@
...
 import { useMedia } from "react-use";
 import FarmingList from "../components/FarmingList";
 
 function FarmingPage() {
-  const campaignData = useLiqudityMiningCampaignData();
+
+  const activeCampaigns = useLiquidityMiningCampaignData("active");
+  const expiredCampaigns = useLiquidityMiningCampaignData("expired");
+  const [campaigns, setCampaigns] = useState({});
+  const [campaignStatus, setCampaignStatus] = useState("active");
```
Comment (Mi-Lan) [inline src/pages/FarmingPage.js:22]: I think we should have and enum for active and inactive campaign and reuse it across the boars so its not hardcoded...

Comment (Mi-Lan) [reply src/pages/FarmingPage.js:22]: also to merge it with options...

File: src/pages/FarmingPage.js [hunk 4]
```diff
@@ -1,19 +1,43 @@
...
 import Search from "../components/Search";
+import DropdownSelect from "../components/DropdownSelect";
 import { useMedia } from "react-use";
 import FarmingList from "../components/FarmingList";
 
 function FarmingPage() {
-  const campaignData = useLiqudityMiningCampaignData();
+
+  const activeCampaigns = useLiquidityMiningCampaignData("active");
+  const expiredCampaigns = useLiquidityMiningCampaignData("expired");
```
Comment (Mi-Lan) [inline src/pages/FarmingPage.js:20]: I think the rest of the code looks fine....Im just not sure if it's best option to have 2 queries like this 🤔  ? Not sure what would alternative be though....

File: src/pages/FarmingPage.js [hunk 5]
```diff
@@ -53,8 +48,8 @@ function FarmingPage() {
           {!below800 && <Search small={true} />}
         </RowBetween>
         <DropdownSelect
-          options={options}
-          active={options[campaignStatus]}
+          options={["Active Campaigns", "Expired Campaigns"]}
+          active={campaignStatus === STATUS.ACTIVE ? "Active Campaigns" : "Expired Campaigns"}
```
Comment (Mi-Lan) [inline src/pages/FarmingPage.js:52]: One suggestion...I think here we should use the STATUS ENUM you created... Other than that looks good!

Comment (natonnelier) [reply src/pages/FarmingPage.js:52]: @Mi-Lan do you mean like?:
```
STATUS: {
  ACTIVE: {
   key: "active",
   message: "Active campaigns"
}
```

The thing is STATUS is an import from `PairData.js` and I thought it didn't make sense to have "message" there.

We could:
- merge messages with STATUS enum in `FarmingPage.js` -> still requires creating a new object
- create a new object like { status: "message" }

What do you think?

Comment (Mi-Lan) [reply src/pages/FarmingPage.js:52]: Ohh no need I just looked at it again...My bad you can't reuse it there.All good!

File: src/contexts/PairData.js [hunk 6]
```diff
@@ -485,7 +485,6 @@ const getPairChartData = async (client, pairAddress) => {
       dayIndexArray.push(data[i]);
       dayData.dailyVolumeUSD = parseFloat(dayData.dailyVolumeUSD);
       dayData.reserveUSD = parseFloat(dayData.reserveUSD);
-      dayData.utilization = parseFloat(dayData.reserveUSD === 0 ? 0 : dayData.dailyVolumeUSD / dayData.reserveUSD) * 100;
```
Comment (Mi-Lan) [inline src/contexts/PairData.js:485]: why was this deleted? This is from pr that was merged?
